### PR TITLE
Dismiss an Inbox Note from `DELETE /wc-analytics/admin/notes/delete/{{id}}` - Networking layer

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #5952

### Description
As part of the Home Screen Updates Milestone 3 (Inbox) project, in this PR I added the networking layer and the unit tests for dismissing an Inbox Note from `DELETE /wc-analytics/admin/notes/delete/{{id}}`. Plus, I added the new `InboxNoteMapper` and the unit tests.

### Testing instructions
Just CI, the code is still not used in the codebase.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Remote/InboxNotesRemote.swift
Comment (Ecarrion) [inline Networking/Networking/Remote/InboxNotesRemote.swift]:
```suggestion
        enqueue(request, mapper: mapper, completion: completion)
```

**Review by Ecarrion — APPROVED** [review]
🚀 x2

**Comment by pmusolino** [comment]
Thank you Ernesto! Before merging this, I'll wait for the approval of this PR https://github.com/woocommerce/woocommerce-ios/pull/6022.

**Pull request merged**
